apps/notifications/src/doc/note-rendering.md

```diff
@@ -0,0 +1,105 @@
+# Notification Detail Rendering
+
+When a user clicks a note to view, that notification is rendered to the user in a slide-out drawer. The pipeline for rendering this drawer to the user begins in `/src/panel/templates/index.jsx`. When the user has selected a note to view, this component renders a child `Note` component from `src/panel/templates/note.jsx` with the property `detailView` set to `true`.
+
+This `detailView` prop further causes the `Note` component to render (among other things) a `NoteBody` component (`/src/panel/templates/body.jsx`). The `render` method for this component is where the majority of rendering logic for a notification takes place.
+
+## Rendering Pipeline
+
+The notification object returned from the server contains, among other things, a `body` array of one or more `block` objects which contain a type, the text of the notification, and an array of `ranges`:
+
+```js
+"ranges": [
+	{
+		"url": "https://www.example.com/",
+		"indices": [
+			397,
+			416
+		]
+	},
+	{
+		"type": "match",
+		"indices": [
+			598,
+			616
+		]
+	},
+	{
+		"type": "match",
+		"indices": [
+			682,
+			700
+		]
+	}
+]
+```
+
+The indices of these ranges indicate locations in the body's text string where special formatting is needed.
+
+After some initial processing, the `NoteBody` component performs a switch statement on each block object based on its type:
+
+```js
+switch ( block.signature.type ) {
+	case 'user':
+		body.push(
+			<User
+				key={ blockKey }
+				block={ block.block }
+				noteType={ this.props.note.type }
+				note={ this.props.note }
+				timestamp={ this.props.note.timestamp }
+				url={ this.props.note.url }
+			/>
+		);
+		break;
+	case 'comment':
+		body.push( <Comment key={ blockKey } block={ block.block } meta={ this.props.note.meta } /> );
+		break;
+	case 'post':
+		body.push( <Post key={ blockKey } block={ block.block } meta={ this.props.note.meta } /> );
+		break;
+	case 'reply':
+		replyBlock = <ReplyBlock key={ blockKey } block={ block.block } />;
+		break;
+	default:
+		body.push( p( html( block.block ) ) );
+		break;
+}
+```
+
+Depending on the item's type, a different block component is rendered - `User` for a representation of a user (for instance, in a notification of a user liking a particular post), `Comment` for a comment made on a blog post, `Post` for a blog post (i.e. on a blog the user is following), `ReplyBlock` for a reply to a comment(?).
```

Comment:
This seems self-documenting to me, but that's just how I see it

Reply:
Eh, maybe a little, but I see some value in further clarifying (particularly for `User`, where the scenario in which a user block might come up doesn't seem entirely intuitive just from the name).

Reply:
> `ReplyBlock` for a reply to a comment(?)

Correct!